### ISSUE TITLE
Version 1.5.0 crashes

### ISSUE THREAD

**Issue by ppaulojr** [datatheorem/TrustKit#133]
I'm seeing a crash upon upgrading to version 1.5.0

```
=============
2017-07-14 15:19:23.520 Orchid[54288:3449502] === TrustKit: Proxy-ing NSURLSessionDelegate: ANSUploader
2017-07-14 15:19:23.658 Orchid[54288:3449496] -[TSKNSURLSessionDelegateProxy URLSession:didBecomeInvalidWithError:]: unrecognized selector sent to instance 0x600000620840
2017-07-14 15:19:23.699 Orchid[54288:3449496] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TSKNSURLSessionDelegateProxy URLSession:didBecomeInvalidWithError:]: unrecognized selector sent to instance 0x600000620840'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000112a0ab0b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000011246f141 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000112a7a134 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x0000000112991840 ___forwarding___ + 1024
	4   CoreFoundation                      0x00000001129913b8 _CF_forwarding_prep_0 + 120
	5   CFNetwork                           0x0000000113a4d01a __42-[NSURLSession finalizeDelegateWithError:]_block_invoke + 39
	6   Foundation                          0x0000000111f793b7 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	7   Foundation                          0x0000000111f790bb -[NSBlockOperation main] + 101
	8   Foundation                          0x0000000111f77877 -[__NSOperationInternal _start:] + 627
	9   Foundation                          0x0000000111f735fc __NSOQSchedule_f + 198
	10  libdispatch.dylib                   0x000000011458405c _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x000000011456294f _dispatch_queue_serial_drain + 221
	12  libdispatch.dylib                   0x0000000114563669 _dispatch_queue_invoke + 1084
	13  libdispatch.dylib                   0x0000000114565ec4 _dispatch_root_queue_drain + 634
	14  libdispatch.dylib                   0x0000000114565bef _dispatch_worker_thread3 + 123
	15  libsystem_pthread.dylib             0x000000011491b5a2 _pthread_wqthread + 1299
	16  libsystem_pthread.dylib             0x000000011491b07d start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException

```

Using AFNetworking 3.0.4

Have you seen this before?

**Comment by nabla-c0d3** [comment]
Hello,
Just to confirm, you switched from 1.4.2 to 1.5.0 and you have swizzling enabled?

**Comment by ppaulojr** [comment]
Yes I do have swizzling enabled

```objc
	kTSKSwizzleNetworkDelegates: @YES,
```

**Comment by adamkaplan** [comment]
I wonder if this is a case of excessive swizzling interactions. Trying to think of a situation where `NSURLSession` has already determined that it's delegate implements that optional method, and then TrustKit swizzles it out.

**Comment by nabla-c0d3** [comment]
It has to be related to a change made after 1.4.2; we barely touched the swizzling code tho, but I will take a look.

**Comment by adamkaplan** [comment]
Nothing of substance in those swizzle proxy classes, at least, that I can see.

https://github.com/datatheorem/TrustKit/compare/1.4.2...1.5.0#diff-3490b759c890bf60e28566b692c7c172

**Comment by nabla-c0d3** [comment]
The `forwardingTargetForSelector:` got removed from the proxies:
https://github.com/datatheorem/TrustKit/compare/1.4.2...1.5.0#diff-3490b759c890bf60e28566b692c7c172L158

**Comment by nabla-c0d3** [comment]
@ppaulojr can you try the latest changes and let me know if it works? Thanks!

**Comment by nabla-c0d3** [comment]
I made some testing myself; this has been released as part of v1.5.1.
